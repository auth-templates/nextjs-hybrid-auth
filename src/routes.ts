export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}