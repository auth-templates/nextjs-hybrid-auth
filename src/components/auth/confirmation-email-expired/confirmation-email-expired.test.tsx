import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ConfirmationEmailExpired from './confirmation-email-expired';
import { PublicRoutes } from '../../routes';

describe("ConfirmationEmailExpired", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <ConfirmationEmailExpired />
            </MemoryRouter>
        );

        expect(screen.getByText(/Your account confirmation email has expired. Please go to/)).toBeInTheDocument(); 
        expect(screen.getByRole('link', {name: 'Resend confirmation email'})).toHaveAttribute('href', PublicRoutes.resendConfirmationEmail);
        expect(screen.getByText(/page in order to send a new mail./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ConfirmationEmailExpired /> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});