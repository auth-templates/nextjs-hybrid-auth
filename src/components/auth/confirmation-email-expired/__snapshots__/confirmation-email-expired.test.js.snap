// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationEmailExpired renders correctly 1`] = `
<div
  className="confirmationEmailExpired"
>
  Your account confirmation email has expired. Please go to
  <a
    href="/resend-confirmation-email"
    onClick={[Function]}
  >
     Resend confirmation email 
  </a>
  page in order to send a new mail.
</div>
`;
